modelkit/core/model.py

```diff
@@ -392,15 +388,18 @@ def _validate(
     ):
         if model:
             try:
-                if self.service_settings.enable_validation:
-                    return model(data=item).data
-                else:
-                    return construct_recursive(model, data=item).data
-            except pydantic.error_wrappers.ValidationError as exc:
-                raise exception(
-                    f"{self.__class__.__name__}[{self.configuration_key}]",
-                    pydantic_exc=exc,
-                ) from exc
+                return model(data=item).data
+            except pydantic.ValidationError as exc:
+                try:
+                    # pydantic 2.* does not implicitely convert
```

Comment:
Is this something we want to continue supporting ?

I am wondering if it is not an opportunity for modelkit to become more explicit about those costly conversions (like pydantic is doing) and let the user do the model_dump if it can afford it or pass directly the good ItemType in input.

Reply:
Fair point. I would tend to minimize breaking changes as much as possible for the migration, but rather add a deprecation warning to inform the user that modelkit will soon stop this implicit conversion, following pydantic 2's steps. What do you think ? This would make the migration easier and help users spot where to add the model_dump manually.

Reply:
I agree not to take breaking changes lightly. And avoid them altogether if not necessary.

However, it is easier to bundle breaking changes in the same major release migration, than doing several migrations I believe. If we delay it to another major release, we take the risk to delay this again and again while it is actually very correlated to the other breaking changes of pydantic that will anyway force the users to question all pydantic models creation.

So accepting this breaking change now is safer for the client and is bringing immediate value by allowing user to question sub-optimal patterns.